edly-io/h5pxblock
# feat: add translations of `es_419` and `es_ES`

### Description
This PR adds translations of `es_419` and `es_ES`  to the H5P XBlock.

### How To Test
1. Install this branch in your environment.
2. Change the platform language in `(Account → Site Preferences → Site Language)`. Set it to "Spanish (Latin America)"
3. Add a H5P block in a unit from Studio.
4. You should see the block in the selected language.

### Using translations commands

A `Makefile` was created that includes the necessary commands [(based on the edX cookie-cutter)](https://github.com/openedx/edx-cookiecutters/blob/master/cookiecutter-xblock/%7B%7Bcookiecutter.repo_name%7D%7D/Makefile) to extract and compile the translations.

1. Run `make extract_translations`, this will update the strings to be translated in `h5pxblock/locale/en/LC_MESSAGES/text.po`
2. Create a folder for your translations (If it does not exist) in `h5pxblock/locale/`, ie: `h5pxblock/locale/es_419/LC_MESSAGES/`, and create your `text.po` file, based on the file generated above.
3. Add all translations for strings according to language.
4. Run `make compile_translations`, this will generate the `text.mo` file in all created language folders.

### Screenshots

1. In the LMS
![image](https://github.com/edly-io/h5pxblock/assets/64033729/b179b935-fd4b-49e3-9303-36409416f86f)
2. In Studio
![image](https://github.com/edly-io/h5pxblock/assets/64033729/d6c1ecad-4c10-4839-858c-0ed9bf0b8adc)
![image](https://github.com/edly-io/h5pxblock/assets/64033729/71aaa0f8-0a3f-4a72-a0a7-5f63925c2afa)
![image](https://github.com/edly-io/h5pxblock/assets/64033729/43b10fb4-b172-4edf-8868-b602c67fed01)
![image](https://github.com/edly-io/h5pxblock/assets/64033729/0d0af106-f668-43fb-bbc9-8de559ce1d96)
